src/charm.py

```diff
@@ -239,75 +217,24 @@ def _cookie_encryption_key(self) -> Optional[str]:
 
     @property
     def _log_level(self) -> str:
-        return self.config["log_level"]
+        return self.config.get("log_level")
 
     @property
     def _domain_url(self) -> Optional[str]:
         return normalise_url(self.ingress.url) if self.ingress.is_ready() else None
```

Comment:
I think that in the rest of the refactored charms we have a class called `PublicIngressData` that is used to get the URL, maybe we should use the same logic here.

Reply:
I deleted the class following Shu's comment, although it didn't have this logic in it.
It seems pretty simple to be ok here, but I also don't have strong opinions here